### PR TITLE
Enable most clj-kondo linters that are disabled by default

### PR DESCRIPTION
**Pull request by vemv**
## Brief

* Enable most clj-kondo linters that are disabled by default
  * One can learn about them in https://github.com/clj-kondo/clj-kondo/blob/v2021.12.19/doc/linters.md
* Upgrade various dependencies
  * It is important to keep our CI representative of how an end-user dep tree should look like.
  * Also keeps antq and nvd-clojure mostly happy.
  * I didn't update Timbre as 5.0.0 seems to involve some breaking changes.

## QA plan

I ran `lein install` and verified that the `reduce` linter is reported:

Screen Shot 2021-12-25 at 12 40 58 PM

## Author checklist

* [x] I have QAed the functionality
* [x] The PR has a reasonably reviewable size and a meaningful commit history
* [x] I have run the [branch formatter](https://github.com/nedap/formatting-stack/blob/332a419034ab46fad526a5592f4257353bd695b6/src/formatting_stack/branch_formatter.clj) and observed no new/significative warnings
* [x] The build passes
* [x] I have self-reviewed the PR prior to assignment
* Additionally, I have code-reviewed iteratively the PR considering the following aspects in isolation:
  * [x] Correctness
  * [x] Robustness (red paths, failure handling etc)
  * [x] Test coverage
  * [x] Spec coverage
  * [x] Documentation
  * [x] Security
  * [x] Performance
  * [x] Breaking API changes
  * [x] Cross-compatibility (Clojure/ClojureScript)

## Reviewer checklist

* [ ] I have checked out this branch and reviewed it locally, running it
* [ ] I have QAed the functionality
* [ ] I have reviewed the PR
* Additionally, I have code-reviewed iteratively the PR considering the following aspects in isolation:
  * [ ] Correctness
  * [ ] Robustness (red paths, failure handling etc)
  * [ ] Test coverage
  * [ ] Spec coverage
  * [ ] Documentation
  * [ ] Security
  * [ ] Performance
  * [ ] Breaking API changes
  * [ ] Cross-compatibility (Clojure/ClojureScript)

### CODE REVIEW

File: src/formatting_stack/linters/kondo.clj [hunk 1]
```diff
@@ -9,23 +9,31 @@
...
-               :unused-private-var   off ;; can give false positives
-               :unresolved-var       off ;; already offered by clj
-               :consistent-alias     off ;; already offered by how-to-ns
-               :duplicate-require    off ;; already offered by clean-ns
-               :unused-import        off ;; already offered by clean-ns
-               :unused-namespace     off ;; already offered by clean-ns
-               :unused-referred-var  off ;; already offered by clean-ns
-               :unresolved-namespace off ;; already offered by clean-ns
+   ;; https://github.com/clj-kondo/clj-kondo/blob/v2021.12.19/doc/linters.md
+   :linters   {:docstring-no-summary                  warn ;; disabled by default
```
Comment (vemv) [inline src/formatting_stack/linters/kondo.clj:18]: Probably the only controversial one would be `:docstring-no-summary`. I agree with the Clojure Style Guide that complete sentences lead to more comprehensible code.

Feel free to deliberate

**Review by vemv — COMMENTED** [review]
Will be a draft pr until https://github.com/clj-kondo/clj-kondo/issues/1514 is solved

File: src/formatting_stack/linters/kondo.clj (hunk 1 above)
Comment (vemv) [reply src/formatting_stack/linters/kondo.clj]: Didn't add :docstring-no-summary finally, it might be feasible if it adds some extra options

**Comment by vemv** [comment]
Ready!

**Review by neominik — APPROVED** [review]
Enabled linters and updated dependencies seem reasonable. Formatting still works as expected.

**Pull request merged**